pool.go

```diff
@@ -489,6 +490,13 @@ func (p *pool) fetchBlockAndUpdate(ctx context.Context, node string, c cid.Cid,
 
 func (p *pool) fetchResourceAndUpdate(ctx context.Context, node string, path string, attempt int, cb DataCallback) (err error) {
 	rm, err := p.fetchResource(ctx, node, path, "application/vnd.ipld.car", attempt, cb)
+	if err != nil && rm.ConnFailure {
+		rm, err = p.fetchResource(ctx, node, path, "application/vnd.ipld.car", attempt, cb)
+		if err == nil {
+			saturnConnectionRetriesTotalMetric.WithLabelValues(resourceTypeCar).Add(1)
+		}
+	}
```

Comment:
I continue to think this is not useful.

We have `attempt` already and are doing this in a loop. having one `attempt` involve sometimes blocking on a single node twice is counter-intuitive and hard to reason about.

Reply:
Removed this.